tests/modules/og_test/og_test.module

```diff
@@ -41,3 +44,23 @@ function og_test_entity_insert(EntityInterface $entity) {
     $membership->save();
   }
 }
+
+/**
+ * Implements hook_og_user_access_entity_operation_alter().
+ */
+function og_test_og_user_access_entity_operation_alter(AccessResultInterface $access_result, CacheableMetadata $cacheable_metadata, $context): void {
+  if (\Drupal::state()->get('og_test_group_content_entity_operation_access_alter', FALSE)) {
+    // Moderators should have access to edit and delete all comments in all
+    // groups.
+    /** @var \Drupal\Core\Session\AccountProxyInterface $user */
+    $user = $context['user'];
+    $group_content = $context['group_content'];
+
+    $is_comment = $group_content->getEntityTypeId() === 'comment';
+    $user_can_moderate_comments = $user->hasPermission('edit and delete comments in all groups');
+
+    if ($is_comment && $user_can_moderate_comments) {
+      $access_result = new AccessResultAllowed();
```

Comment:
This would probably work too but it looks ugly:
$access_result->orIf(new AccessResultAllowed());